packages/smart-contracts/src/contracts/lib/SafeERC20.sol

```diff
@@ -90,4 +83,23 @@ library SafeERC20 {
     /* solium-enable security/no-inline-assembly */
     return result;
   }
+
+  
+  bytes4 private constant SELECTOR = bytes4(keccak256(bytes('transfer(address,uint256)')));
+
+  function safeTransfer(IERC20 _token, address _to, uint256 _amount) internal {
+      address tokenAddress = address(_token);
+
+      // check if the address is a contract
+      assembly {
+        if iszero(extcodesize(tokenAddress)) { revert(0, 0) }
+      }
+      
+      (bool success, bytes memory data) = tokenAddress.call(abi.encodeWithSelector(
+          SELECTOR, 
+          _to, 
+          _amount
+      ));
+      require(success && (data.length == 0 || abi.decode(data, (bool))), 'Transfer failed');
```

Comment:
Is this replacing what we're doing with assembly on the other functions on this library?
If we're sure it's the same it's great, and we should drop the assembly from the others.

Reply:
It's how uniswap dealt with USDT, fully tested but I cannot say if it's 100% the same as the assembly lines. What I can do is remove the assembly lines of the 2 others, and test (with automated tests but also playing around).

Reply:
From my understanding it looks the same. But I may be missing something.
If this can replace the assembly it would be great.

Reply:
Done 👍